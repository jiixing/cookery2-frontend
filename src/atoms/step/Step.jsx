// @flow

import React from 'react'

import * as R from 'ramda'
import { Link } from 'react-router-dom'

const renderSimpleStep = ({ description }: Object) => (
  <div>
    <span>{description}</span>
  </div>
)

const renderReferenceStep = ({ id, name }: Object) => (
  <Link to={`/recipes/${id}`}>
    {name}
  </Link>
)

const Step = (props: Object) => (
  <span>
    {props.isRecipe ? renderReferenceStep(R.pick(['name', 'id'])(props)) : renderSimpleStep(R.pick(['description'])(props))}
  </span>
)

export default Step
