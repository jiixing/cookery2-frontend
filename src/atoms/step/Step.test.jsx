import React from 'react'
import { fullRender } from 'test'

import Step from './Step'

describe('components', () => {
  describe('Step', () => {
    it('renders correctly for simple steps', () => {
      const { component } = fullRender(<Step isRecipe={false} description='This is just a step' />)
      expect(component.toJSON()).toMatchSnapshot()
    })

    it('renders correctly for reference steps', () => {
      const { component } = fullRender(<Step isRecipe name='name of the step' id={3} />)
      expect(component.toJSON()).toMatchSnapshot()
    })
  })
})
