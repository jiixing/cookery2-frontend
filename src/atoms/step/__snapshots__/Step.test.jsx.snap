// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Step renders correctly for reference steps 1`] = `
<span>
  <a
    href="/recipes/3"
    onClick={[Function]}
  >
    name of the step
  </a>
</span>
`;

exports[`components Step renders correctly for simple steps 1`] = `
<span>
  <div>
    <span>
      This is just a step
    </span>
  </div>
</span>
`;
